File: Algorithms/1111.maximum-nesting-depth-of-two-valid-parentheses-strings/maximum-nesting-depth-of-two-valid-parentheses-strings.cpp
class Solution {
public:
    vector<int> maxDepthAfterSplit(string seq) {
        //1，use stack
        int d = 0;
        vector<int> ans;
        for (char& c : seq)
            if (c == '(') {
                ++d;  //增加深度
                ans.push_back(d % 2);
            }
            else {
                ans.push_back(d % 2);
                --d;  //减少深度
            }
        return ans;
    }
};

File: Algorithms/1111.maximum-nesting-depth-of-two-valid-parentheses-strings/maximum-nesting-depth-of-two-valid-parentheses-strings.go
func maxDepthAfterSplit(seq string) []int {
    //1,use stack
    depth := 0
    var ans []int
    for _,c := range seq {
        if c == '(' {
            depth++
            ans = append(ans,depth % 2)
        } else {
            ans = append(ans,depth % 2)
            depth--
        }
    }
    return ans
}
